src/reference/asciidoc/kafka.adoc

```diff
@@ -326,3 +326,23 @@ public void listen(@Payload String foo,
     ...
 }
 ----
+
+===== Handling Duplicates
+
+In certain scenarios, such as rebalancing, a message may be redelivered that has already been processed.
+The framework cannot know whether or not such a message has been processed or not, that is an application-level
+function.
+This is known as the http://www.enterpriseintegrationpatterns.com/patterns/messaging/IdempotentReceiver.html[Idempotent
+Receiver] pattern and Spring Integration provides an http://docs.spring.io/spring-integration/reference/html/messaging-endpoints-chapter.html#idempotent-receiver
+[implementation thereof].
+
+The Spring for Apache Kafka project also provides some assistance by means of the `DeduplicatingMessageListenerAdapter`
+and `DeduplicatingAcknowledgingMessageListenerAdapter` classes, which can wrap your `MessageListener` or
+`AcknowledgingMessageLister` respectively.
+Each of these classes takes an implementation of `DeDuplicationStrategy` where
+you implement the `isDuplicate` method to signal that a message is a duplicate and should be discarded.
+
+Similarly, when using `@KafkaListener`, the `DeduplicationStrategy` can be injected into the container factory.
+
+Of course, if you are using manual acknowledgments you should not use these classes because you would need to
+acknowledge the duplicated message.
```

Comment:
I don't understand this phrase.
We have `AcknowledgingMessageListener` which definitely assumes a `manual` acknowledgment. So, according this your conclusion the `DeduplicatingAcknowledgingMessageListenerAdapter` is a bad class.
Or we should : 
```
else {
	acknowledgment.acknowledge();
}
```
in case of duplicate anyway there...